packages/react/src/common/is-empty.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Check if a value is empty
+ * @param {unknown} value - Value to check
+ * @returns {boolean} True if the value is empty
+ */
+export function isEmpty(value: unknown): boolean {
+  if (value === null || value === undefined) {
+    return true;
+  }
+
+  if (typeof value === 'string' && value.trim() === '') {
+    return true;
+  }
+
+  if (Array.isArray(value) && value.length === 0) {
+    return true;
+  }
+
+  if (typeof value === 'object' && Object.keys(value).length === 0) {
+    return true;
+  }
+
+  return false;
+}
```

Comment:
I know that this is just for checking if the `changedFlags` are empty but why are we checking for empty strings here? It feels quite generic for the concrete use case as we are just interested in `changedFlags` being `undefined` or empty `string[]` right? [[1]](https://github.com/open-feature/js-sdk/blob/e6adae04a5f3481317a855968ad846c72af4a087/packages/shared/src/events/eventing.ts#L26 )

And if I am overseeing something here, I am not sure about the semantic of `isEmpty` here as e.g. lodash defines   `isEmpty` differently regarding the non enumerable values [[2]](https://github.com/lodash/lodash/blob/6a2cc1dfcf7634fea70d1bc5bd22db453df67b42/src/isEmpty.ts#L72 ) but this might be fine, just want to bring it up.

[[1]](https://github.com/open-feature/js-sdk/blob/e6adae04a5f3481317a855968ad846c72af4a087/packages/shared/src/events/eventing.ts#L26 ): https://github.com/open-feature/js-sdk/blob/e6adae04a5f3481317a855968ad846c72af4a087/packages/shared/src/events/eventing.ts#L26
[[2]](https://github.com/lodash/lodash/blob/6a2cc1dfcf7634fea70d1bc5bd22db453df67b42/src/isEmpty.ts#L72 ): https://github.com/lodash/lodash/blob/6a2cc1dfcf7634fea70d1bc5bd22db453df67b42/src/isEmpty.ts#L72

Reply:
Yeah, I agree. I'll try to find a cleaner way to handle this.